src/app/components/import-annotations/add-annotations/add-annotations.component.ts

```diff
@@ -0,0 +1,364 @@
+  retrieveResolvers,
+} from "@baw-api/resolver-common";
+import { TagsService } from "@baw-api/tag/tags.service";
+import { ErrorCardStyle } from "@shared/error-card/error-card.component";
+import { BawApiError } from "@helpers/custom-errors/baw-api-error";
+import { isInstantiated } from "@helpers/isInstantiated/isInstantiated";
+import { INTERNAL_SERVER_ERROR } from "http-status";
+import {
+  addAnnotationImportMenuItem,
+  annotationsImportCategory,
+} from "../import-annotations.menu";
+import { annotationImportRoute } from "../import-annotations.routes";
+
+const audioEventImportKey = "audioEventImport";
+
+@Component({
+  selector: "baw-add-annotations",
+  templateUrl: "add-annotations.component.html",
+})
+class AddAnnotationsComponent
+  extends PageComponent
+  implements OnInit, UnsavedInputCheckingComponent
+{
+  public constructor(
+    private api: AudioEventImportFileService,
+    private tagsApi: TagsService,
+    private route: ActivatedRoute,
+    private router: Router,
+    private notifications: ToastrService,
+    @Inject(ASSOCIATION_INJECTOR) private injector: AssociationInjector
+  ) {
+    super();
+  }
+
+  @ViewChild(NgForm) private form: NgForm;
+
+  /** The route model that the annotation import is scoped to */
+  public audioEventImport: AudioEventImport;
+
+  protected errorCardStyles = ErrorCardStyle;
+
+  /**
+   * A boolean value indicating if the currently buffered audio event import can
+   * be successfully committed
+   */
+  protected valid: boolean = false;
+  protected uploading: boolean = false;
+  protected importFiles: File[] = [];
+  protected additionalTagIds: Id[] = [];
+
+  /**
+   * Errors that apply to the entire file that we cannot display in the events
+   * table.
+   * E.g. duplicate file uploads, names, etc...
+   */
+  protected importErrors: BawErrorData[] = [];
+  // we use a BehaviorSubject for the audio event import files because users can upload
+  // multiple files through the file input and new subscribers should get the most recent value
+  protected importFiles$ = new BehaviorSubject<AudioEventImportFile[]>([]);
+
+  public get hasUnsavedChanges(): boolean {
+    return this.hasFiles || this.hasAdditionalTags;
+  }
+
+  // if the "Import Annotations" button is disabled, we want to provide some
+  // feedback to the user outlining why they cannot submit the form using a
+  // tooltip
+  protected get uploadTooltip(): string | null {
+    // by returning null, the tooltip will not be displayed
+    if (this.canCommitUploads()) {
+      return null;
+    }
+
+    if (!this.hasFiles) {
+      return "Please select a file to upload";
+    } else if (!this.valid || !this.form.valid) {
+      return "Please fix all errors before submitting";
+    } else if (this.uploading) {
+      return "Please wait for the current upload to complete";
+    }
+
+    // we should never hit this condition, but if we do, I return null so that
+    // we don't end up in an unknown state
```

Comment:
nit: concise comments are less load

```suggestion
    // should never hit, but be safe
```